File: test/fixtures/python/post_json_multiple_headers.py
import requests

headers = {
    'Content-Type': 'application/json',
    'X-API-Version': '200',
}

json_data = {
    'userName': 'username123',
    'password': 'password123',
    'authLoginDomain': 'local',
}

response = requests.post('http://localhost:28139/rest/login-sessions', headers=headers, json=json_data, verify=False)

# Note: json_data will not be serialized by requests
# exactly as it was in the original request.
#data = '{"userName":"username123","password":"password123", "authLoginDomain":"local"}'
#response = requests.post('http://localhost:28139/rest/login-sessions', headers=headers, data=data, verify=False)

File: test/fixtures/matlab/post_json_multiple_headers.m
%% Web Access using Data Import and Export API
% This is not possible with the webread/webwrite API

%% HTTP Interface
import matlab.net.*
import matlab.net.http.*
import matlab.net.http.io.*

header = [
    HeaderField('Content-Type', 'application/json')
    HeaderField('X-API-Version', '200')
]';
uri = URI('http://localhost:28139/rest/login-sessions');
options = HTTPOptions('VerifyServerName', false);
body = JSONProvider(struct(...
    'userName', 'username123',...
    'password', 'password123',...
    'authLoginDomain', 'local'...
));
response = RequestMessage('post', header, body).send(uri.EncodedURI, options);
